### PR TITLE
update workflow with more automation

### LINKED ISSUE #1257: Update Release Process Automation
Now that we have #813, we can update the release process to be more automated.

Summary: Instead of needing to do version update and benchmarks manually, it happens automatically upon the pushing of a release branch.

## General New Outline
- Push a release branch to origin `release=v#.#.#`
- Kicks off workflow
- Workflow runs benchmarks on everything and commits to that branch (Skip for pre-release branches)
- Workflow runs `make version v={tag}` and commits to update the version
- Workflow adds the version tag
- Kicks off creation of the GH release and artifacts

### PR DESCRIPTION
**Pull request by demisx**
# Goal
The goal of this PR is to further improve release workflow to allow releases to be cut with pushing a release tag only. 

Closes #1257

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -7,13 +7,125 @@ on:
     tags:
       - "v[0-9]+.[0-9]+.[0-9]+" # ex. v1.0.0
       - "v[0-9]+.[0-9]+.[0-9]+-rc[0-9]+" # ex. v1.1.0-rc1
-      - "vX.X.X" # used for testing only
+      - "v0.0.1" # used for testing only
+      - "v0.0.1-rc[0-9]+" # used for testing only
```
[the remark this answers is not in the document]
Comment (wilwade) [reply .github/workflows/release.yml:11]: We should block docker and npm push with these tags so we do not get latest broken somehow.

File: Makefile [hunk 2]
```diff
@@ -226,20 +228,27 @@ POLKADOT_VERSION=$(shell awk -F "=" '/name = "polkadot-cli"/,/version = ".*"/{ p
...
-	find ./ -type f -name 'Cargo.toml' -exec sed -i '' 's/^version = \"0\.0\.0\"/version = \"$(v)+polkadot$(POLKADOT_VERSION)\"/g' {} \;
-	$(MAKE) check
+ifeq ($(UNAME), Linux)
+	$(eval $@_SED := -i -e)
+endif
+ifeq ($(UNAME), Darwin)
+	$(eval $@_SED := -i '')
+endif
+	find . -type f -name "Cargo.toml" -print0 | xargs -0 sed $($@_SED) 's/^version = \"0\.0\.0\"/version = \"$(v)+polkadot$(POLKADOT_VERSION)\"/g';
+	@echo "Doing cargo check with fake examples project seems to be the easiest way to update version in Cargo.lock"
```
Comment (wilwade) [inline Makefile:250]: These are doc examples we are checking not a fake project. There just are very few doc examples (currently none).
```suggestion
	@echo "Doing cargo check for just examples seems to be the easiest way to update version in Cargo.lock"
```

**Review by wilwade — CHANGES_REQUESTED** [review]
Mostly just one change to make sure that test releases do not push to docker or npm.

**Review by saraswatpuneet — APPROVED** [review]
That update paired well this PR :)

**Pull request merged**